### PR TITLE
[GEN-2112]: update RBAC permissions for Collector

### PR DESCRIPTION
**Pull request by BenElferink**
This pull request includes several changes to the `Makefile`, `README.md`, and various Go files related to RBAC roles and permissions. The most important changes involve adding comments for clarity, updating the `deploy` target in the `Makefile`, and documenting RBAC permissions for different components.

### Updates to `Makefile`:

* Modified the `deploy` target to ensure all components are deployed sequentially using `make` commands.

### Documentation Improvements:

* Added a detailed section in `cli/cmd/resources/README.md` to track RBAC permissions requested by different components, organized by `APIGroups`, `Resources`, `Verbs`, and the `Component` requesting them.

### Codebase Simplification:

* Added comments in `cli/cmd/resources/autoscaler.go` to explain the purpose of each `PolicyRule` in the `NewAutoscalerRole` function.
* Added comments in `cli/cmd/resources/datacollection.go` to clarify the purpose of each `PolicyRule` in the `NewDataCollectionClusterRole` function.
* Added comments in `cli/cmd/resources/odiglet.go` to describe the purpose of each `PolicyRule` in the `NewOdigletRole` function.

### CODE REVIEW

**Comment by BenElferink** [comment]
@blumamir this PR is now ready for review 👍

File: cli/cmd/resources/datacollection.go [hunk 1]
```diff
@@ -34,51 +34,33 @@ func NewDataCollectionClusterRole(psp bool) *rbacv1.ClusterRole {
...
-					"get",
-					"list",
-					"watch",
-				},
+			{ // TODO: remove this after we remove honeycomb custom exporter config
+				// located at: autoscaler/controllers/datacollection/custom/honeycomb.go
 				APIGroups: []string{""},
-				Resources: []string{
-					"pods",
-					"endpoints",
```
Comment (blumamir) [inline cli/cmd/resources/datacollection.go:46]: @tamirdavid1 is it related to the load balancer of the tail sampler maybe?

**Review by blumamir — APPROVED** [review]

File: cli/cmd/resources/datacollection.go (hunk 1 above)
Comment (tamirdavid1) [reply cli/cmd/resources/datacollection.go:46]: Yea seems like indeed the load balancer uses the endpoints
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/loadbalancingexporter/resolver_k8s.go#L105

also it written in this readme.md
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/loadbalancingexporter/example/k8s-resolver/README.md\

@BenElferink @blumamir

Comment (BenElferink) [reply cli/cmd/resources/datacollection.go:46]: No problem, I'll add it again :)

**Pull request merged**